Model/Service/Recommendation/Category.php

```diff
@@ -77,27 +77,22 @@ public function getSortedProductIds(
         if (!$featureAccess->canUseGraphql()) {
             return $productIds;
         }
-
-        switch ($type) {
-            case NostoHelperSorting::NOSTO_PERSONALIZED_KEY:
-                $recoOperation = new CategoryBrowsingHistory($nostoAccount, $nostoCustomerId);
-                break;
-            default:
-                $recoOperation = new CategoryTopList($nostoAccount, $nostoCustomerId);
-                break;
+        if ((string)$type === NostoHelperSorting::NOSTO_PERSONALIZED_KEY) {
+            $recoOperation = new CategoryBrowsingHistory($nostoAccount, $nostoCustomerId);
+        } else {
+            $recoOperation = new CategoryTopList($nostoAccount, $nostoCustomerId);
         }
         $recoOperation->setCategory($category);
         try {
             $result = $recoOperation->execute();
             foreach ($result as $item) {
-                if ($item->getProductId()) {
+                if ($item->getProductId() && is_int($item->getProductId())) {
```

Comment:
How about using `is_numeric` here. Not sure if it was Magento but some platform had product ids in format `123.323`

Reply:
Good point, fixed.